dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreRecordCollection.cs

```diff
@@ -125,6 +125,38 @@ public RedisVectorStoreRecordCollection(IDatabase database, string collectionNam
         }
     }
 
+    /// <inheritdoc />
+    public string CollectionName => this._collectionName;
+
+    /// <inheritdoc />
+    public async Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await this._database.FT().InfoAsync(this._collectionName).ConfigureAwait(false);
+            return true;
+        }
+        catch (RedisServerException ex) when (ex.Message.Contains("Unknown index name"))
```

Comment:
Checking for text inside a message is fragile, guessing this is the best option available.

Is there a way to get access to the message from the Redis library?

Reply:
Unfortunately it's just hardcoded in the service, without any error codes 😞, see https://github.com/RediSearch/RediSearch/blob/master/src/info_command.c#L89